File: budget-management-frontend/src/app/model/interfaces/budgetItem.ts
import { ExpenseCategory } from "../enums/expenseCategory";
import { ExpenseSchedule } from "../enums/expenseSchedule";
import { PaymentMethod } from "../enums/PaymentMethod";

export interface BudgetItem {
    id?: number;
    category?: ExpenseCategory;
    label: string;
    targetAmount: number;
    schedule?: ExpenseSchedule;
    autoAddToExpenses: boolean;
    paymentMethod: PaymentMethod;
    date: Date;
    spent?: number;
}

File: budget-management-backend/src/main/java/com/suika/bm/model/product/BudgetItem.java
package com.suika.bm.model.product;

import com.suika.bm.model.enums.ExpenseCategory;
import com.suika.bm.model.enums.ExpenseSchedule;
import com.suika.bm.model.enums.PaymentMethod;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@ToString
public class BudgetItem {

    Long id;

    ExpenseCategory category;

    String label;

    float targetAmount;

    ExpenseSchedule schedule;

    boolean autoAddToExpenses;

    PaymentMethod paymentMethod;

    LocalDate date;

    Integer spent;
}
